dev-requirements.txt

```
@@ -21,7 +21,8 @@ sphinx-rfcsection~=0.1.1
 # (or until py3.9 EOL... in 10/2025, I HOPE NOT)
 vcrpy @ git+https://github.com/sopel-irc/vcrpy@uncap-urllib3
 # type check
-mypy>=1.3,<2
+# often breaks CI on master, so pin and update deliberately, on our own terms
+mypy==1.11.2
```

Is it acceptable to accept [compatible version](https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release) for patch versions only (i.e. 1.11.3, .4, etc. but not 1.12)?

```suggestion
mypy~=1.11.2
```

---

I don't expect the `mypy` project to release any more 1.11.x versions now that it's moved on to 1.12. Their [PyPI release history](https://pypi.org/project/mypy/#history) is quite linear.

And even "compatible versions" can change something, though I do have a _bit_ more confidence in `mypy`'s versioning than, say, `sphinxcontrib-autoprogram`… I'd feel more comfortable with an exact-version pin, because I'm _very_ tired of merging a green PR only for `master` to go red because something updated.

_With apologies to @SnoopJ: I resolved his thread without commenting, since I would say basically the same thing twice and we don't need that. :)_